corehq/apps/userreports/reports/filters/factory.py

```diff
@@ -86,6 +86,22 @@ def _build_dynamic_choice_list_filter(spec, report):
     )
 
 
+def _build_multi_field_dynamic_choice_list_filter(spec, report):
+    wrapped = MultiFieldDynamicChoiceFilterSpec.wrap(spec)
+    choice_provider_spec = wrapped.get_choice_provider_spec()
+    choice_provider = MultiFieldChoiceProviderFactory.from_spec(choice_provider_spec)(report, wrapped.slug)
+    choice_provider.configure(choice_provider_spec)
+    return DynamicChoiceListFilter(
+        name=wrapped.slug,
+        datatype=wrapped.datatype,
+        field=wrapped.fields,
```

Comment:
Instead of returning `DynamicChoiceListFilter` could you create a `MultiFieldDynamicChoiceLIstFilter` so that this can be `fields`? It would be much less confusing that way